python/pyarrow/_dataset.pyx

```diff
@@ -1237,6 +1238,40 @@ cdef class CsvFileFormat(FileFormat):
         return f"<CsvFileFormat parse_options={self.parse_options}>"
 
 
+# From io.pxi
+def py_buffer(object obj):
```

Comment:
This should be importable as just `from pyarrow.lib import py_buffer` (you can see _plasma.pyx and _cuda.pyx do the same)

Reply:
These duplicates have been removed, once I figured out how to move all relevant parts into `io.pxi`